lib/blocks.js

```diff
@@ -274,32 +271,28 @@ BlockController.prototype.list = function(req, res) {
     async.mapSeries(
       hashes,
       function(hash, next) {
-        self._getBlockSummary(hash, moreTimestamp, next);
+        self._getBlockSummary(hash.blockhash, next);
       },
       function(err, blocks) {
         if(err) {
           return self.common.handleErrors(err, res);
         }
 
-        blocks.sort(function(a, b) {
-          return b.height - a.height;
-        });
-
         var data = {
           blocks: blocks,
           length: blocks.length,
           pagination: {
             next: next,
             prev: prev,
-            currentTs: lte - 1,
+            currentTs: high - 1,
             current: dateStr,
             isToday: isToday,
             more: more
           }
         };
 
         if(more) {
-          data.pagination.moreTs = moreTimestamp;
+          data.pagination.moreTs = hashes[hashes.length - 1].logicalts - 1;
```

Comment:
Since this works as non-inclusive end, we should use:

```
data.pagination.moreTs = hashes[hashes.length - 1].logicalts
```
Instead of:
```
data.pagination.moreTs = hashes[hashes.length - 1].logicalts - 1;
```